### PR TITLE
Emote system minor improvements, Screaming in pain and silencing your victims

### PR DESCRIPTION
**Pull request by MaxIsJoe**
Been a while since I touched this huh?
The new improvements should pave the way for #8328 to come a reality one day as the emote system is no longer just tied to text input, meaning that emotes can finally be called from wherever and whenever from now on.

Removed a check for nonbinary characters as the `default` check already does the same exact thing

Added something that we've wanted to add for a long while now but we've been so busy that me (and I think Gillies as well) haven't got around to adding and it's the ability for players to AAAAAAAAAAAAAAAAAAAAAAAAAAAA when getting attacked, players will automatically now call the scream emote when they've received over 18 damage and have a 20 second cooldown between each automatic scream.

Though sounds have still not been added yet, there will be a time where they will be and traitors and doctors will want to muffle their victims screams of pain so the pass-bying clown doesn't look for the source of the scream or report it. The "gag" trait on mask items like the muzzle will stop players from giving away their position by screaming loudly and disturbing the entire station.

### Changelog:
CL: [New] - Players will now automatically scream when receiving over 18 points of damage.
CL: [New] - The *Scream emote can now be muffled by a muzzle.
CL: [Fix] - Fixed issues where players can use specific emotes while being dead.

### CODE REVIEW

**Comment by MaxIsJoe** [comment]
any reviewers know why `CheckIfPlayerIsGagged` is not working?

**Comment by PerfectTangent** [comment]
> players will automatically now call the scream emote when they've received over 18 damage and have a 20 second cooldown between each automatic scream.

18 total damage on the character or 18 in a single attack?

**Comment by Bod9001** [comment]
rebased to get tests working

**Comment by MaxIsJoe** [comment]
> > players will automatically now call the scream emote when they've received over 18 damage and have a 20 second cooldown between each automatic scream.
> 
> 18 total damage on the character or 18 in a single attack?

18 in a single attack

File: UnityProject/Assets/Scripts/Core/Chat/EmoteActionManager.cs [hunk 1]
```diff
@@ -7,14 +7,19 @@ namespace Core.Chat
 {
 	public class EmoteActionManager : MonoBehaviour
 	{
-		[SerializeField]
-		private List<EmoteSO> emotes;
+		public static EmoteActionManager Instance;
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Core/Chat/EmoteActionManager.cs]: if this is going to be a singleton monobehaviour, why not use the class we made for them?

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Core/Chat/EmoteActionManager.cs]: Rider kept bugging me about it for some reason so I just gave up and did it manually

Comment (NoooneyDude) [reply UnityProject/Assets/Scripts/Core/Chat/EmoteActionManager.cs]: What was it complaining about?

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Core/Chat/EmoteActionManager.cs]: > What was it complaining about?

Something related to a manager namespace? idk i kept trying to find what it wanted but even the quick actions didn't give me anything

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Core/Chat/EmoteActionManager.cs]: rider finally pissed off

**Comment by MaxIsJoe** [comment]
still unsure why the function is not getting called

**Comment by Bod9001** [comment]
it looks okay, Are you sure it's not failing on the first  first part of if statement?

**Comment by MaxIsJoe** [comment]
> it looks okay, Are you sure it's not failing on the first first part of if statement?

Yes I'm sure

**Comment by Bod9001** [comment]
it's because your overriding the logic inside of GenderedEmote so it's, never getting called

**Comment by MaxIsJoe** [comment]
> it's because your overriding the logic inside of GenderedEmote so it's, never getting called

oh shoot, thanks Bod! I've also fixed a bug I found when you pointed this out.

This should be good to go now

**Comment by corp-0** [comment]
was the manager added to online scene? I don't see any scene changes

**Comment by MaxIsJoe** [comment]
> was the manager added to online scene? I don't see any scene changes

wdym?

**Comment by corp-0** [comment]
well the emote manager is a singleton monobehaviour and we normally add those to an scene so it is initiallized and code can actually use it

**Comment by MaxIsJoe** [comment]
does that include the chat object as well?
because it's currently on it still.

**Review by NoooneyDude — APPROVED** [review]

**Review by corp-0 — APPROVED** [review]

**Pull request merged**